### ISSUE TITLE
UI bug – small time interval differences fail to make key show both values on hover

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#30642]
![CleanShot 2022-02-03 at 12 36 35](https://user-images.githubusercontent.com/11967660/152416883-10b0a574-9c82-41a1-9e60-906db620cb75.gif)

Notice that these datapoints have the same time but show only 1 value (not 2 values) in the key because it's treating them as separate time points. 

> Here is an example, these two series have snapshots as follows: "2022-02-03T19:34:22Z" and "2022-02-03T19:34:21Z"

**Comment by Joelkw** [comment]
Arbitrarily, if two points are within 10 minutes of each other, I think we should show the same tooltip (we could do more than 10 minutes as well – the threshold seems the easier part to change than just building this feature).

**Comment by vovakulikov** [comment]
@sourcegraph/code-insights-backend I think this is a good example that we have to do data post-processing on the frontend to solve problems like this. If we have more charts and more data on the page for each insight it could lead to perf problems. 

I think we may have a filter over series data in GQL query to be able to query series points grouped by N minutes/days/...

```gql
series {
 points(grouped: 600) {
   dateTime 
 }
}
```

**Comment by chwarwick** [comment]
I don't know that we would want to group data over an interval of time, because in the current state those two data points on different series are not actually from the same time.  They reflect the difference in the point of time when the search queries were run, if we were to consolidate them into a single point the data would not be accurate. 

I think correct the long term solution is that we need to make the time intervals consistent across the insight view which would mean they would be consistent across the series on the view as well.  Currently when you backfill an insight you will get perfect intervals.  For example a daily insight will run the queries against specific commits so that it can show the data at exactly 1 day intervals, however when new points are added to the series they are based on the time that the process runs which introduces variability into that time interval. The size variability will depend on a number of factors but in some cases on demo I saw it being as much as 3 hours.  We have now lost a consistent series and if we were attempting to do any higher level functionality like rate of change we now introduced a 2nd variable into it.
